lib/validation.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require "time"
+require "netaddr"
 
 module Validation
```

Comment:
It would be beneficial to include separate tests for the validation helper in their commits. Although these are likely covered in the vm API routes, it's not explicitly clear that all cases are accounted for. It's important to clearly confirm that all validation rules are functioning as expected.